### ISSUE TITLE
Fine-tune BERT

### ISSUE THREAD

**Issue by lamthuy** [explosion/spacy-transformers#168]
Hi, 
In the following article:
https://explosion.ai/blog/spacy-transformers
The authors said that: "Pipeline components can communicate gradients back to the transformer by incrementing the doc._.trf_d_last_hidden_state attribute, which is a numpy/cupy array that holds the gradient with respect to the last hidden state of the transformer. To implement a custom component for a new task, you should create a new subclass of spacy.pipeline.Pipe that defines the Model, predict, set_annotations and update methods. During the update method, your component will receive a batch of Doc objects that hold the transformer features, and a batch of GoldParse objects that contain the gold-standard annotations. Using these inputs, you should update your model and increment the gradient for the last-hidden states on the doc._.trf_last_hidden_state variable. Your subclass can handle its model any way you like, but the easiest approach if you're using PyTorch is to use Thinc's PyTorch wrapper class, which will save you from having to implement the to/from bytes/disk serialization methods. We expect to publish a full tutorial with the recommended workflow in future."

Could you please elaborate more on how to fine-tune the Bert's weights?

**Comment by svlandeg** [comment]
The newly released `v1.0.0rc0` version of this library contains built-in components that take care of this backpropagation automatically when you include a `Transformer` as the `tok2vec` sublayer of another component. The gradients will flow back to the Transformer, depending on the classification task(s) that are run on top of the Transformer layer. Did you have any specific questions around that process?

If you'd like to try out the new `v1.0.0rc0`, have a look at the revamped spaCy v3, for which a a pre-release is available as [`spacy-nightly`](https://pypi.org/project/spacy-nightly/). Relevant docs can be found at https://nightly.spacy.io/usage/embeddings-transformers#transformers and https://nightly.spacy.io/api/transformer.
